### ISSUE TITLE
Update for Webpack 4

### ISSUE THREAD

**Issue by odoe** [OpenNTF/dojo-webpack-plugin#146]
Currently the plugin doesn't work with Webpack 4. I'm trying `4.0.1`.
It fails on this section
https://github.com/OpenNTF/dojo-webpack-plugin/blob/master/lib/DojoLoaderPlugin.js#L38

This is the error.

```
Error: Plugin could not be registered at 'get dojo require'. Hook was not found.
BREAKING CHANGE: There need to exist a hook at 'this.hooks'. To create a compatiblity layer for this hook, hook into 'this._pluginCompat'.
```
Looking at [this issue](https://github.com/webpack/webpack/issues/6064) it has breaking change for plugins.

```
new plugin system
plugin method is backward-compatible
Plugins should use Compiler.hooks.xxx.tap(<plugin name>, fn) now
```

There's a [comment in there](https://github.com/webpack/webpack/issues/6064#issuecomment-349405474) that references how a migration from 3 to 4 would look.

I've forked this project, but I was just starting to learn how to write a webpack plugin from the previous guide, so not sure how quickly I could do a PR on this update.

Thanks!

**Comment by chuckdumont** [comment]
Working on it.  There are many breaking changes, and [a fix](https://github.com/webpack/webpack/pull/6674) is needed in webpack.

**Comment by chuckdumont** [comment]
Support for webpack 4 is included in version 2.7.0.  Note that 4.2.0 is required.
